sdk/typespec/typespec_client_core/src/async_runtime/mod.rs

```diff
@@ -61,9 +65,11 @@ pub type SpawnedTask = Pin<
 pub type SpawnedTask =
     Pin<Box<dyn Future<Output = std::result::Result<(), Box<dyn std::error::Error>>> + 'static>>;
 
-/// An async command runner.
+/// An Asynchronous Runtime.
 ///
-pub trait TaskSpawner: Send + Sync + Debug {
+/// This trait defines the various
+///
+pub trait AsyncRuntime: Send + Sync {
```

Comment:
Can't be `Send` on WASM.

Reply:
FWIW, as far as I've been able to figure out, `Send` *is* defined on WASM.

The only thing about `Send` on WASM is that the future returned by a WASM `async fn` doesn't implement `Send`, thus it is incompatible with predefined futures which require `Send` (which is basically all non-WASM futures).

So it's my understanding that `Send` on a trait is fine (it says that an implementation of the trait can be passed between threads), but a function within the trait that returns a future cannot include `Send` in the future's declaration.

Please correct me if my understanding is incorrect.